doc/hardware/porting/soc_porting.rst

```diff
@@ -0,0 +1,324 @@
+.. _soc_porting_guide:
+
+SoC Porting Guide
+###################
+
+To add Zephyr support for a new :term:`SoC`, you at least need a *SoC directory*
+with various files in it. Also a SoC :file:`.dtsi` in
+:zephyr_file:`dts/<ARCH>/<VENDOR>` is required.
+
+SoC Definitions
+***************
+
+It is expected that you are familiar with the board concept in Zephyr.
+A high level overview of the hardware support hierarchy and terms used in the
+Zephyr documentation can be seen in :ref:`hw_support_hierarchy`.
+
+For SoC porting, the most important terms are:
+
+- SoC: the exact system on a chip the board's CPU is part of
+- SoC series: a smaller group of tightly related SoCs
+- SoC family: a wider group of SoCs with similar characteristics
+- CPU Cluster: a cluster of one or more CPU cores.
+- CPU core: a particular CPU in an architecture
+- Architecture: an instruction set architecture
+
+Architecture
+============
+
+See :ref:`architecture_porting_guide`.
+
+
+Create your SoC directory
+*************************
+
+Each SoC must have a unique name. Use the official name given by the SoC vendor
+and check that it's not already in use. In some cases someone else may have
+contributed a SoC with identical name. If the SoC name is already in use, then
+you should probably improve the existing SoC instead of creating a new one.
+The script ``list_hardware`` can be used to retrieve a list of all SoCs known
+in Zephyr, for example ``./scripts/list_hardware.py --soc-root=. --socs`` from
+the Zephyr base directory for a list of names that are already in use.
+We will use ``soc1`` in the following
+
+Start by creating the directory ``zephyr/soc/<VENDOR>/soc1``, where
+``<VENDOR>`` is your vendor subdirectory.
+
+.. note::
+  A ``<VENDOR>`` subdirectory is mandatory if contributing your SoC
+  to Zephyr, but if your SoC is placed in a local repo, then any folder
+  structure under ``<your-repo>/soc`` is permitted.
+  The ``<VENDOR>`` subdirectory must match a vendor defined in the list in
+  :zephyr_file:`dts/bindings/vendor-prefixes.txt`. If the SoC vendor does not
+  have a prefix in that list, then one must be created.
+
+.. note::
+
+  The SoC directory name does not need to match the name of the SoC.
+  Multiple SoC can even defined be in one directory. In Zephyr, SoCs are often
+  organized in sub-folders under a common SoC Family or SoC Series tree.
```

Comment:
nit: 
organized in sub-folders under a common SoC Family or SoC Series tree.
->
organized in sub-folders in a common SoC Family or SoC Series tree.
(I guess, if it is a tree it would be "in")

Reply:
fixed